### PR TITLE
Authorization scheme

### LINKED ISSUE #23: Support different scheme for authorization
The goal here would be that given a certain `image_path`:
1. Get the user from the backend, through the cookie
2. Verify that `user.slurm_account` is set - if not, fail
3. Verify that `image_path` is relative to `{ZARR_DATA_BASE_PATH}/{user.slurm_account}/` - if not, fail

For the moment we can support two different schemes, and switch via another env variable `AUTHORIZATION_SCHEME="allowed-list", "user-folders"`.

### PR DESCRIPTION
**Pull request by zonia3000**
## Checklist before merging
- [X] I added an appropriate entry to `CHANGELOG.md`

Closes #23

### CODE REVIEW

**Comment by zonia3000** [comment]
@tcompa I've refactored a bit the code to have configuration loaded in one file and authorization checks performed in another file. I've also added some unit tests for the authorization logic and configured CI to run them.

**Comment by tcompa** [comment]
Should we rename `ALLOWED_USERS` into `ALLOWED_USERS_FILE` or `ALLOWED_USER_PATH`?

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  // Cookie cache TTL in seconds
+  const cacheExpirationTime = process.env.CACHE_EXPIRATION_TIME ? parseInt(process.env.CACHE_EXPIRATION_TIME) : 60;
+
+  let basePath = process.env.BASE_PATH || '/vizarr';
+  if (!basePath.endsWith('/')) {
+    basePath += '/';
+  }
+
+  const allowedUsers = getAllowedUsers(allowedUsersFile);
```
[the remark this answers is not in the document]
Comment (zonia3000) [reply src/config.ts]: I moved it as you suggested. Thanks

**Comment by zonia3000** [comment]
> Should we rename `ALLOWED_USERS` into `ALLOWED_USERS_FILE` or `ALLOWED_USER_PATH`?

It is now called `ALLOWED_USERS_FILE`.

**Pull request merged**